main/webapp/modules/core/scripts/index/default-importing-controller/controller.js

```diff
@@ -265,10 +265,13 @@ Refine.DefaultImportingController.prototype.getPreviewData = function(callback,
           "limit" : numRows || 100 // More than we parse for preview anyway
         }),
         null,
-        function(data) {
```

Comment:
This is the callback for a $.POST with jQuery. But, where is the failure callback?

See this example:
https://github.com/OpenRefine/OpenRefine/blob/577caabec1cce968df56156d7561a1ffde4b01c4/main/webapp/modules/core/scripts/preferences.js#L85-L91

Here, the error from the API is handled. At the very least, a dialog should be displayed. But maybe a move back in the interface also…

Reply:
Sure there needs to be a faliure callback . To gain more insight on error handling on callbacks and how to implement it I have searched the code base but there seems to be no code the hanldles  callback errors. Could I be missing something ?

Reply:
> @antoine2711: Sure there needs to be a faliure callback . To gain more insight on error handling on callbacks and how to implement it I have searched the code base but there seems to be no code the hanldles callback errors. Could I be missing something ?

@WaltonG: there is at least ONE place that handles the errors of API well, and it's in the setPrefs and getPrefs. I know because I coded it. And yes, you are right to point that it is missing in a lot of places.

Now, I could ask you to fix many (all of them?) in code surrounding yours, but It would take a lot of time. What I want here is to make sure that you know at least how to write it ONCE. That's my mentor work. And you have learned enough for me to ask you this level of quality. ;-) Keep up the good work!

Regards, Antoine